md2conf/application.py

```diff
@@ -15,16 +23,70 @@ def update_document(
     api.update_page(document.page_id, document.xhtml())
 
 
+
+def synchronize_pages(api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions) -> None:
+    if os.path.isdir(path):
+        synchronize_directory(api, path, options)
+    elif os.path.isfile(path):
+        synchronize_page(api, path, options)
+    else:
+        raise ValueError(f"expected: valid file of directory path. got: {path}")
+
+
 def synchronize_page(
-    api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions
+    api: ConfluenceSession, path: str, options: ConfluenceDocumentOptions, page_metadata: dict[str, str] = dict()
```

Comment:
This package is marked as Python 3.8-compatible in [PyPI](https://pypi.org/project/markdown-to-confluence/). As such, I am afraid we must stick to `typing.Dict` for annotating dictionary types.

Reply:
Makes sense!